marimo/_plugins/ui/_impl/tables/ibis_table.py

```diff
@@ -186,14 +203,103 @@ def get_bin_values(
                     ),
                 ).order_by("bin")
 
+                return result.execute()
+
+            def get_bin_values(
+                self, column: str, num_bins: int
+            ) -> list[BinValue]:
+                """Get bin values for a column. Currently supports numeric and temporal columns.
+
+                Args:
+                    column (str): The column to get bin values for.
+                    num_bins (int): The number of bins to create.
+
+                Returns:
+                    list[BinValue]: The bin values.
+                """
+                if column not in self.data.columns:
+                    LOGGER.error(f"Column {column} not found in Ibis table")
+                    return []
+
+                col = self.data[column]
+                dtype = col.type()
+
+                if dtype.is_temporal():
+                    return self._get_bin_values_temporal(
+                        column, dtype, num_bins
+                    )
+
+                if not dtype.is_numeric():
+                    return []
+
+                bin_values = self._get_numeric_bin_values(col, num_bins)
+
                 return [
                     BinValue(
                         bin_start=row.bin_start,
                         bin_end=row.bin_end,
                         count=row.count,
                     )
-                    for row in result.execute().itertuples(index=False)
+                    for row in bin_values.itertuples(index=False)
+                ]
+
+            def _get_bin_values_temporal(
```

Comment:
we can also leave off ibis since we are soon moving it to be managed by narwhals. if you already have it, thats great, but if there is anymore work, no need to since it will get "done" when we move it to narwhals